src/south/ProtocolFactory.class.js

```diff
@@ -0,0 +1,40 @@
+const protocolList = {}
+protocolList.ADS = require('./ADS/ADS.class')
+protocolList.Modbus = require('./Modbus/Modbus.class')
+protocolList.OPCUA_HA = require('./OPCUA_HA/OPCUA_HA.class')
+protocolList.OPCUA_DA = require('./OPCUA_DA/OPCUA_DA.class')
+protocolList.MQTT = require('./MQTT/MQTT.class')
+protocolList.SQLDbToFile = require('./SQLDbToFile/SQLDbToFile.class')
+protocolList.FolderScanner = require('./FolderScanner/FolderScanner.class')
+protocolList.OPCHDA = require('./OPCHDA/OPCHDA.class')
```

Comment:
I'm confused about this ProtocolFactory class with 2 static methods which tends to make me think it's not a class. 

It appears that it should methods from the baseEngine class (in this case, the engine would not need to be passed to create (as it would be "this").

Reply:
The idea of using the factory design patterns in this was to reduce the complexity of the Engines and the number of imports. But the BaseEngine is, indeed, not so complex, so it can be moved there.